nl2fetch.py

```diff
@@ -22,6 +22,8 @@ def __init__(self, ip: str, port: int):
         self.nl2.connect()
         self.last_pos: Tuple[int, int, int] = None
         """Initializes a variable to track the last position of the roller-coaster cart. This will be used to calculate linear velocity over each frame of the simulation. Each value in the tuple represents the x, y, and z position of the roller-coaster cart."""
+        self.play_mode = False
```

Comment:
Nit: Could potentially rename this variable to something like `self.started_receiving_packets` to make the `elif` later more readable. `self.play_mode` just feels a little confusing considering we also use `data.in_play_mode`.